File: 백준/Silver/1080. 행렬/행렬.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine()," ");
		
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		
		int[][] A = makeMatrix(br, n, m);
		int[][] B = makeMatrix(br, n, m);
		
		int maxY = n - 3 + 1;
		int maxX = m - 3 + 1;
		int count = 0;
		
		for(int y = 0; y < maxY; y++) {
			for(int x = 0; x< maxX; x++) {
				if(A[y][x] != B[y][x]) {
					changeMatrix(A, x, y);
					count++;
				}
			}
		}
		
		System.out.println(compareTwoMatrix(A, B) ? count : -1);
	}
	static int[][] makeMatrix(BufferedReader br, int height, int width) throws IOException{
		int[][] matrix = new int[height][width];
		for(int y = 0; y < height; y++) {
			char[] charArray = br.readLine().trim().toCharArray();
			
			for(int x = 0; x < width; x++) {
				matrix[y][x] = charArray[x] - '0';
			}
		}
		return matrix;
	}
	static void changeMatrix(int[][] matrix, int startX, int startY) {
		int endY = Math.min(startY + 3, matrix.length);
		int endX = Math.min(startX + 3, matrix[0].length);
		
		for(int y = startY; y < endY; y++) {
			for(int x = startX; x < endX; x++) {
				matrix[y][x] = (matrix[y][x] + 1) % 2;
			}
		}
	}
	static boolean compareTwoMatrix(int[][] matrixA, int[][] matrixB) {
		for(int y = 0; y < matrixA.length; y++) {
			for(int x = 0; x < matrixA[0].length; x++) {
				if(matrixA[y][x] != matrixB[y][x]) {
					return false;
				}
			}
		}
		return true;
	}
}

File: 백준/Silver/1080. 행렬/행렬.py
from sys import stdin

n, m = map(int, stdin.readline().split())

A = [list(stdin.readline().rstrip()) for _ in range(n)]
B = [list(stdin.readline().rstrip()) for _ in range(n)]

matrix_A = [[int(A[y][x]) for x in range(len(A[0]))] for y in range(len(A))]
matrix_B = [[int(B[y][x]) for x in range(len(B[0]))] for y in range(len(B))]

def compare_two_matrix(matrix_a, matrix_b):
    for y in range(len(matrix_a)):
        for x in range(len(matrix_a[0])):
            if matrix_a[y][x] != matrix_b[y][x]:
                return False

    return True

def change_matrix(matrix, start_x, start_y):
    end_y = min(len(matrix), start_y + 3)
    end_x = min(len(matrix[0]), start_x + 3)
    for y in range(start_y, end_y):
        for x in range(start_x, end_x):
            matrix[y][x] = (matrix[y][x] + 1) % 2

max_y = n - 3 + 1
max_x = m - 3 + 1
count = 0
for y in range(max_y):
    for x in range(max_x):
        if matrix_A[y][x] != matrix_B[y][x]:
            change_matrix(matrix_A, x, y)
            count += 1

print(count if compare_two_matrix(matrix_A, matrix_B) else -1)